src/QuadForm/ClosestVector.jl

```diff
@@ -373,13 +382,26 @@ function closest_vectors(L::ZLat, v::Vector{RingElement} , upperbound::RingEleme
                 push!(cv, x)
             end
             V = ambient_space(L)
-            # debug remove later when performance matters
-            for x in cv
-                t = x - v
-                dist = inner_product(V,t,t)
-                @assert dist<= upperbound
+            cv2 = Array{Array{fmpz,1},1}()
+            if bool==false
+                for x in cv
+                    t = x - v
+                    dist = inner_product(V,t,t)
+                    if dist <= upperbound
+                        push!(cv2,x)
+                    end
+                end
+                return sort!(cv2)
```

Comment:
Please remove the `!sort` here and below. It is not necessary to sort the results.

Reply:
If I remove the ```sort!```, then most of my tests would fail because the results I get from Shimada's algorithm are not generating results in the same order as the ones using ```shortest_vector```. Should I maybe add another optional argument to the function to compute sorted vs. unsorted list of vectors?

Reply:
seems good.